src/metabase/upload.clj

```diff
@@ -678,7 +676,54 @@
 ;;; |  hydrate based_on_upload for FE
 ;;; +--------------------------------
 
-(mi/define-simple-hydration-method based-on-upload
+(defn uploadable-table-ids
+  "Returns the subset of table ids where the user can upload to the table."
+  [table-ids]
+  (if (empty? table-ids)
+    #{}
+    (let [tables (t2/hydrate (t2/select :model/Table :id [:in table-ids]) :db)]
```

Comment:
nit: I think this would have been clearer with a threading macro: 

```clojure
(->> (t2/select ..)
      t2/hydrate
      (keep ...)
      set)
```